pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -674,6 +733,56 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         _require(tokenData != 0, Errors.INVALID_TOKEN);
     }
 
+    // Join/exit callbacks
+
+    function _beforeJoinExit(
+        uint256[] memory,
+        uint256[] memory,
+        uint256
+    ) internal virtual override {
+        _collectAumManagementFees();
+    }
+
+    /**
+     * @dev Calculates the AUM fees accrued since the last collection and pays it to the pool manager.
+     * This function is called automatically on joins and exits.
+     */
+    function _collectAumManagementFees() internal {
+        uint256 lastCollection = _lastAumFeeCollectionTimestamp;
+        uint256 currentTime = block.timestamp;
+
+        // Collect fees based on the time elapsed
+        if (currentTime > lastCollection) {
+            // Reset the collection timer to the current block
+            _lastAumFeeCollectionTimestamp = currentTime;
+
+            uint256 managementAumFeePercentage = getManagementAumFeePercentage();
+
+            // `lastCollection == 0` means that we're in the first attempt to collect AUM fees
+            // For gas reasons we only collect AUM fees from this point onwards so perform an early return if so.
+            // We also perform an early return if pool's emergency pause mechanism has been triggered.
+            if (managementAumFeePercentage == 0 || lastCollection == 0 || !_isNotPaused()) {
+                return;
+            }
+
+            uint256 elapsedTime = currentTime - lastCollection;
+            // Similar to BPT swap fee calculation, collect fees equal to the AUM % after minting
+            // F is the AUM fee percentage, S is the totalSupply, and x is the amount to mint after 1 year:
+            // S + x = S + F(S + x)
+            // x = F(S + x)
+            // x(1 - F) = FS
+            // x = S * F/(1 - F); per annual time period
```

Comment:
I found this hard to understand and had to go back to the explanation of BPT fees in WeightedMath: instead of mentioning how this is similar to that scenario, I'd instead have here a longer-form explanation like the one that exists on that file.

Reply:
I've added a longer explanation in [9be13c5](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1164/commits/9be13c5b36520c8537ce76fc82bf78995e97b749)